cpp/src/communicator/ucxx.cpp

```diff
@@ -415,13 +423,18 @@ class SharedResources {
         for (auto& callback : delayed_progress_callbacks)
             callback();
         if (!worker_->isProgressThreadRunning()) {
+            // TODO: Support blocking progress mode in addition to polling
             worker_->progress();
-            // TODO: Support blocking progress mode
         }
 
         clear_completed_futures();
     }
 
+    void maybe_progress_worker() {
+        if (++progress_count % 100)
+            progress_worker();
```

Comment:
Why the `100` ?
```suggestion
        if (++progress_count % 100) { progress_worker(); }
```

Reply:
No specific reason, I just borrowed the `100` from [OpenMPI's default](https://github.com/open-mpi/ompi/blob/7ad7adad676773fc61203e9a536d17b2ebdfa9c8/opal/mca/common/ucx/common_ucx.c#L42).

Reply:
Add that in a comment

Reply:
Added in https://github.com/rapidsai/rapidsmpf/pull/487/commits/783d2d10d73a1f07429b5eb43171ea01ab9aa493